### PR TITLE
long f = (long) Float.NaN

### PR DESCRIPTION
**Pull request by SergeyLabutin**
detect problem

### CODE REVIEW

File: jtransc-rt/resources/d/Base.d [hunk 1]
```diff
@@ -236,8 +236,28 @@ class N {
...
 
-	static public long f2j(float v) { return cast(long)(v); }
-	static public long d2j(double v) { return cast(long)(v); }
+	static public long f2j(float v) {
+		if (std.math.isNaN(v)) {
+			return 0;
+		} else if (!std.math.isInfinity(v)) {
+			return cast(long)v;
+		} else if (v >= 0) {
+			return cast(long)0x7fffffffffffffff;
```
Comment (intrigus) [inline jtransc-rt/resources/d/Base.d]: Is there no constant for the largest 64bit integer? 
Same for all other languages.

File: jtransc-rt/resources/d/Base.d [hunk 2]
```diff
@@ -236,8 +236,28 @@ class N {
 
 	static public long i2j(int v) { return v; }
 
-	static public long f2j(float v) { return cast(long)(v); }
-	static public long d2j(double v) { return cast(long)(v); }
+	static public long f2j(float v) {
+		if (std.math.isNaN(v)) {
```
Comment (intrigus) [inline jtransc-rt/resources/d/Base.d]: Is there a reason why the most likely case is not the first?

Comment (SergeyLabutin) [inline jtransc-rt/resources/d/Base.d]: !std.math.isInfinity(v) && !std.math.isNaN(v) i think no different, but i no known dart or d languages.

File: jtransc-rt/resources/d/Base.d (hunk 1 above)
Comment (SergeyLabutin) [inline jtransc-rt/resources/d/Base.d]: i will fix it

File: jtransc-rt/resources/d/Base.d (hunk 2 above)
Comment (SergeyLabutin) [inline jtransc-rt/resources/d/Base.d]: for faster need function isFinity(), but have languages it?

Comment (soywiz) [inline jtransc-rt/resources/d/Base.d]: I had the same issue. Some languages doesn't have it. Just have isInfinity, which is false for nan

**Comment by SergeyLabutin** [comment]
@soywiz please see problem with testTime, testNanoTime, i can not understand this test.

**Comment by soywiz** [comment]
@SergeyLabutin I have downloaded the branch. I'm testing it to check what's wrong.

**Pull request merged**